src/tfr_control/include/tfr_control/drivebase_publisher.h

```diff
@@ -0,0 +1,49 @@
+/****************************************************************************************
+ * File:            drivebase_publisher.h
+ * 
+ * Purpose:         This class is used by the drivebase node to convert from a Twist
+ *                  message in /cmd_vel to angular velocity for the two motor sets
+ *                  (left and right), handling the calculations for differential steering.
+ * 
+ *                  Every message read from /cmd_vel will result in two messages sent,
+ *                  one to each motor controller. This class will only publish after
+ *                  reading a message on /cmd_vel.
+ * 
+ * Subscribed To:   /cmd_vel
+ * Publishes To:    /left_tread_velocity_controller/command
+ *                  /right_tread_velocity_controller/command
+ ***************************************************************************************/
+#ifndef DRIVEBASE_PUBLISHER_H
+#define DRIVEBASE_PUBLISHER_H
+
+#include "ros/ros.h"
+#include "geometry_msgs/Twist.h"
+
+namespace tfr_control
+{
+    class DrivebasePublisher
+    {
+    public:
+        DrivebasePublisher(ros::NodeHandle& n, float axle_length, float wheel_radius);
+        DrivebasePublisher(const DrivebasePublisher& other) = delete;
```

Comment:
Explicitly deleting things is good, but if we are going to do it you should delete the other generated methods.

This is the stuff that would get generated by a c++11 compiler
class Thing {
public:
  //not generated in your case Thing();                        // default constructor
    Thing(const Thing&);            // copy c'tor
    Thing& operator=(const Thing&); // copy-assign
//your discretion    ~Thing();                       // d'tor
    // C++11:
    Thing(Thing&&);                 // move c'tor
    Thing& operator=(Thing&&);      // move-assign
};

Reply:
good article on generated methods in c++ 11
https://stackoverflow.com/questions/3734247/what-are-all-the-member-functions-created-by-compiler-for-a-class-does-that-hap

good article on asking whether you would want to delete a destructor
https://stackoverflow.com/questions/9091957/explicitly-deleting-destructors-and-not-calling-delete

Reply:
Good idea. Mostly I just wanted to give a hint to someone down the line who might invoke the copy constructor that we probably don't want to have multiple copies of this object, since the constructor interacts with the ROS infrastructure, subscribing to topics and stuff like that.